src/Build/BackEnd/Components/Caching/ResultsCache.cs

```diff
@@ -350,9 +350,14 @@ private static bool CheckResults(BuildResult result, List<string> targets, HashS
         /// <param name="buildResult">The candidate build result.</param>
         /// <returns>True if the flags and project state filter of the build request is compatible with the build result.</returns>
         private static bool AreBuildResultFlagsCompatible(BuildRequest buildRequest, BuildResult buildResult)
-        {
+        { 
+            if (buildResult.BuildRequestDataFlags is null)
```

Comment:
Could you please explain this scenario? 
Why when the request data flags is null results that there are equal?

Reply:
Absent `BuildRequestDataFlags` data means it was not read from a results cache, we need to fall back to the previous behavior when this information was not available. From [this line](https://github.com/dotnet/msbuild/blob/c7904f283c2a5d4490916e09c4db6fe546c27e3d/src/Build/BackEnd/Components/Caching/ResultsCache.cs#L175) it is clear that we should consider the flags equal as this is behavior prior PR #9987